packages/cli/lib/sandboxes.js

```diff
@@ -0,0 +1,179 @@
+
+function getAccountName(config) {
+  const isSandbox =
+    config.sandboxAccountType && config.sandboxAccountType !== null;
+  const sandboxName = `[${getSandboxType(config.sandboxAccountType)} sandbox] `;
+  return `${config.name} ${isSandbox ? sandboxName : ''}(${config.portalId})`;
+}
+
+async function getSyncTypes(parentAccountConfig, config) {
+  const parentPortalId = parentAccountConfig.portalId;
+  const portalId = config.portalId;
+  const syncTypes = await fetchTypes(parentPortalId, portalId);
+  return syncTypes.map(t => ({ type: t.name }));
+}
+
+const sandboxCreatePersonalAccessKeyFlow = async (env, account, name) => {
+  const configData = await personalAccessKeyPrompt({ env, account });
+  const updatedConfig = await updateConfigWithPersonalAccessKey(configData);
+
+  if (!updatedConfig) {
+    process.exit(EXIT_CODES.SUCCESS);
+  }
+
+  let validName = updatedConfig.name;
+
+  if (!updatedConfig.name) {
+    const nameForConfig = name
+      .toLowerCase()
+      .split(' ')
+      .join('-');
+    const { name: promptName } = await enterAccountNamePrompt(nameForConfig);
+    validName = promptName;
+  }
+
+  updateAccountConfig({
+    ...updatedConfig,
+    environment: updatedConfig.env,
+    tokenInfo: updatedConfig.auth.tokenInfo,
+    name: validName,
+  });
+  writeConfig();
+
+  const setAsDefault = await setAsDefaultAccountPrompt(validName);
+
+  logger.log('');
+  if (setAsDefault) {
+    logger.success(
+      i18n(`cli.lib.prompts.setAsDefaultAccountPrompt.setAsDefaultAccount`, {
+        accountName: validName,
+      })
+    );
+  } else {
+    const config = getConfig();
+    logger.info(
+      i18n(`cli.lib.prompts.setAsDefaultAccountPrompt.keepingCurrentDefault`, {
+        accountName: config.defaultPortal,
+      })
+    );
+  }
+  logger.success(
+    i18n('cli.commands.sandbox.subcommands.create.success.configFileUpdated', {
+      configFilename: DEFAULT_HUBSPOT_CONFIG_YAML_FILE_NAME,
+      authMethod: PERSONAL_ACCESS_KEY_AUTH_METHOD.name,
+      account: validName,
+    })
+  );
+  uiFeatureHighlight([
+    'accountsUseCommand',
+    'accountOption',
+    'accountsListCommand',
+  ]);
+};
+
+const ACTIVE_TASK_POLL_INTERVAL = 1000;
+
+const isTaskComplete = task => {
+  if (!task) {
+    return false;
+  }
+  return task.status === 'COMPLETE';
+};
+
+function pollSyncStatus(accountId, taskId) {
+  const i18nKey = 'cli.commands.sandbox.subcommands.sync.types';
+  const multibar = new cliProgress.MultiBar(
+    {
+      hideCursor: true,
+      format: '[{bar}] {percentage}% | {taskType}',
+    },
+    cliProgress.Presets.rect
+  );
+  const mergeTasks = {
+    'lead-flows': 'forms', // Since UI only shows forms, merge lead-flows into forms progress
+  };
+  const barInstances = {};
+  return new Promise((resolve, reject) => {
+    const pollInterval = setInterval(async () => {
+      const taskResult = await fetchTaskStatus(accountId, taskId).catch(reject);
+      if (taskResult.tasks) {
```

Comment:
This conditional is quite confusing. Is there any way to simplify this code?

Reply:
Let me add some more comments here to clarify things a bit, if its still too confusing I'd be happy to zoom whenever!

Reply:
Sounds good to me!